xlators/storage/posix/src/posix-common.c

```diff
@@ -373,18 +391,40 @@ posix_reconfigure(xlator_t *this, dict_t *options)
                " fallback to <hostname>:<export>");
     }
 
+    old_disk_reserve = priv->disk_reserve;
     GF_OPTION_RECONF("reserve", priv->disk_reserve, options, percent_or_size,
                      out);
     /* option can be any one of percent or bytes */
     priv->disk_unit = 0;
     if (priv->disk_reserve < 100.0)
         priv->disk_unit = 'p';
 
-    if (priv->disk_reserve) {
+    if (old_disk_reserve && !priv->disk_reserve)
+    {
+        /* Delete a pxl object from a list of disk_reserve was enabled
+           earlier now it is disabled
+        */
+        pxl = priv->pxl;
+        if (pxl) {
+            pthread_mutex_lock(&ctx->xl_lock);
+            {
+                pxl->detach_notify = _gf_true;
+                while (pxl->is_use)
+                    pthread_cond_wait(&pxl->cond, &ctx->xl_lock);
+
+                list_del_init(&pxl->list);
+                priv->pxl = NULL;
+            }
+            pthread_mutex_unlock(&ctx->xl_lock);
+            free(pxl);
+        }
```

Comment:
Create a function for this. This is exact the same code used in `posix_notify()`.

We should also decrement the `diskxl_count` and join the thread if it's 0. Otherwise we may have issues when enabling/disabling this multiple times.